code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -573,3 +573,183 @@ CIGARETTE PACKETS ARE IN FANCY.DM
+			emagged = 1
+			super = 0
+			user << "<span class='warning'>You maximize the voltage in the [src]</span>"
+			add_overlay(image(icon, "vapeopen_high"))
+			var/datum/effect_system/spark_spread/sp = new /datum/effect_system/spark_spread //for effect
+			sp.set_up(5, 1, src)
+			sp.start()
+		else
+			user << "<span class='warning'>The [name] is already emagged!</span>"
+	else
+		user << "<span class='notice'>You need to open the cap to do that</span>"
+
+/obj/item/clothing/mask/cigarette/vape/attack_self(mob/user)
+	if(reagents.total_volume > 0)
+		user << "<span class='notice'>you empty [src] of all reagents.</span>"
+		smoketime = 0
+		reagents.clear_reagents()
+	return
+
+/obj/item/clothing/mask/cigarette/vape/equipped(mob/user, slot)
+	if(slot == slot_wear_mask)
+		if(!screw)
+			user << "<span class='notice'>You start puffing on that dank vape</span>"
+			reagents.set_reacting(TRUE)
+			lit = 1
+			START_PROCESSING(SSobj, src)
+			light("<span class='notice'>[user] starts puffing on that fine vape</span>")
+		else //it will not start if the vape is opened.
+			user << "<span class='warning'>You need to close the cap first!</span>"
+
+/obj/item/clothing/mask/cigarette/vape/dropped(mob/user)
+	var/mob/living/carbon/C = user
+	if(C.get_item_by_slot(slot_wear_mask) == src)
+		lit = 0
+		reagents.set_reacting(FALSE)
+		STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/mask/cigarette/vape/handle_reagents()//had to rename to avoid duplicate error
+	if(reagents.total_volume)
+		if(iscarbon(loc))
+			var/mob/living/carbon/C = loc
+			if (src == C.wear_mask) // if it's in the human/monkey mouth, transfer reagents to the mob
+				var/fraction = min(REAGENTS_METABOLISM/reagents.total_volume, 1) //this will react instantly, making them a little more dangerous than cigarettes
+				reagents.reaction(C, INGEST, fraction)
+				reagents.trans_to(C, REAGENTS_METABOLISM)
+				if(reagents.get_reagent_amount("welding_fuel"))
+					//HOT STUFF
+					C.fire_stacks = 2
+					C.IgniteMob()
+
+				if(reagents.get_reagent_amount("plasma")) // the plasma explodes when exposed to fire
+					var/datum/effect_system/reagents_explosion/e = new()
+					e.set_up(round(reagents.get_reagent_amount("plasma") / 2.5, 1), get_turf(src), 0, 0)
+					e.start()
+					if(ismob(loc))
+						var/mob/M = loc
+						M.unEquip(src, 1)
+					qdel(src)
+				return
+		reagents.remove_any(REAGENTS_METABOLISM)
+
+/obj/item/clothing/mask/cigarette/vape/process()
+	var/mob/living/M = loc
+
+	if(isliving(loc))
+		M.IgniteMob()
+
+	vapetime++
+
+	if(!reagents.total_volume)
+		if(ismob(loc))
+			M << "<span class='notice'>The [name] is empty!</span>"
+			STOP_PROCESSING(SSobj, src)
+			//it's reusable so it won't unequip when empty
+		return
+	//open flame removed because vapes are a closed system, they wont light anything on fire
+
+	if(super && vapetime > 3)//Time to start puffing those fat vapes, yo.
+		var/datum/effect_system/smoke_spread/chem/s = new
+		s.set_up(reagents, 1, loc)

```

Comment:
I have reservations about how OP this is (smoke clouds multiply the initial reagents), so just standing in your own smoke cloud could be extremely effective. Also, the cloud lasts a bit too long to be a good pick for vaping puffs.